File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/DependencyModelBuilder.java
package io.qimia.uhrwerk.common.model;

import org.jetbrains.annotations.NotNull;

public class DependencyModelBuilder extends StateModelBuilder<DependencyModelBuilder> {

  Long id;
  Long tableId;
  Long dependencyTargetId;
  Long dependencyTableId;
  TableModel table;
  TableModel dependencyTable;
  TargetModel dependencyTarget;
  String area;
  String vertical;
  String tableName;
  String format;
  String version;
  PartitionTransformType transformType;
  PartitionUnit transformPartitionUnit;
  int transformPartitionSize;

  public DependencyModelBuilder id(Long id) {
    this.id = id;
    return this;
  }

  public DependencyModelBuilder tableId(@NotNull Long tableId) {
    this.tableId = tableId;
    return this;
  }

  public DependencyModelBuilder dependencyTargetId(@NotNull Long dependencyTargetId) {
    this.dependencyTargetId = dependencyTargetId;
    return this;
  }

  public DependencyModelBuilder dependencyTableId(@NotNull Long dependencyTableId) {
    this.dependencyTableId = dependencyTableId;
    return this;
  }

  public DependencyModelBuilder table(TableModel table) {
    this.table = table;
    return this;
  }

  public DependencyModelBuilder dependencyTable(TableModel dependencyTable) {
    this.dependencyTable = dependencyTable;
    return this;
  }

  public DependencyModelBuilder dependencyTarget(TargetModel dependencyTarget) {
    this.dependencyTarget = dependencyTarget;
    return this;
  }

  public DependencyModelBuilder area(String area) {
    this.area = area;
    return this;
  }

  public DependencyModelBuilder vertical(String vertical) {
    this.vertical = vertical;
    return this;
  }

  public DependencyModelBuilder tableName(String tableName) {
    this.tableName = tableName;
    return this;
  }

  public DependencyModelBuilder format(String format) {
    this.format = format;
    return this;
  }

  public DependencyModelBuilder version(String version) {
    this.version = version;
    return this;
  }

  public DependencyModelBuilder transformType(PartitionTransformType transformType) {
    this.transformType = transformType;
    return this;
  }

  public DependencyModelBuilder transformPartitionUnit(PartitionUnit transformPartitionUnit) {
    this.transformPartitionUnit = transformPartitionUnit;
    return this;
  }

  public DependencyModelBuilder transformPartitionSize(int transformPartitionSize) {
    this.transformPartitionSize = transformPartitionSize;
    return this;
  }

  public DependencyModel build() {
    return new DependencyModel(this);
  }

  @Override
  DependencyModelBuilder getThis() {
    return this;
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/builders/DependencyModelBuilder.kt
package io.qimia.uhrwerk.common.metastore.builders

import io.qimia.uhrwerk.common.metastore.model.DependencyModel
import io.qimia.uhrwerk.common.metastore.model.PartitionTransformType
import io.qimia.uhrwerk.common.metastore.model.PartitionUnit
import io.qimia.uhrwerk.common.metastore.model.TableModel
import io.qimia.uhrwerk.common.model.*

class DependencyModelBuilder : StateModelBuilder<DependencyModelBuilder>() {
    var id: Long? = null
    var tableId: Long? = null
    var dependencyTargetId: Long? = null
    var dependencyTableId: Long? = null
    var table: TableModel? = null
    var dependencyTable: TableModel? = null
    var dependencyTarget: TargetModel? = null
    var area: String? = null
    var vertical: String? = null
    var tableName: String? = null
    var viewName: String? = null
    var format: String? = null
    var version: String? = null
    fun id(id: Long?): DependencyModelBuilder {
        this.id = id
        return this
    }

    fun tableId(tableId: Long): DependencyModelBuilder {
        this.tableId = tableId
        return this
    }

    fun dependencyTargetId(dependencyTargetId: Long): DependencyModelBuilder {
        this.dependencyTargetId = dependencyTargetId
        return this
    }

    fun dependencyTableId(dependencyTableId: Long): DependencyModelBuilder {
        this.dependencyTableId = dependencyTableId
        return this
    }

    fun table(table: TableModel?): DependencyModelBuilder {
        this.table = table
        return this
    }

    fun dependencyTable(dependencyTable: TableModel?): DependencyModelBuilder {
        this.dependencyTable = dependencyTable
        return this
    }

    fun dependencyTarget(dependencyTarget: TargetModel?): DependencyModelBuilder {
        this.dependencyTarget = dependencyTarget
        return this
    }

    fun area(area: String?): DependencyModelBuilder {
        this.area = area
        return this
    }

    fun vertical(vertical: String?): DependencyModelBuilder {
        this.vertical = vertical
        return this
    }

    fun tableName(tableName: String?): DependencyModelBuilder {
        this.tableName = tableName
        return this
    }

    fun viewName(viewName: String?): DependencyModelBuilder {
        this.viewName = viewName
        return this
    }

    fun format(format: String?): DependencyModelBuilder {
        this.format = format
        return this
    }

    fun version(version: String?): DependencyModelBuilder {
        this.version = version
        return this
    }


    fun build(): DependencyModel {
        return DependencyModel(
            id,
            tableId,
            dependencyTargetId,
            dependencyTableId,
            table,
            dependencyTable,
            dependencyTarget,
            area,
            vertical,
            tableName,
            viewName,
            format,
            version
        )
    }

    override fun getThis() = this
}